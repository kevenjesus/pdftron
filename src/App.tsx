import { useEffect, useRef } from 'react'
import './App.css'
import WebViewer from '@pdftron/webviewer';

function App() {
  const viewer = useRef(null);

  useEffect(() => {
    if (viewer.current) {
      WebViewer(
        {
          // path: `${import.meta.env.VITE_BASE_URL}webviewer/lib`,
          path: `/webviewer/lib`,
          licenseKey: import.meta.env.VITE_PDFTRON_LICENSE_KEY,
          initialDoc: 'https://pdftron.s3.amazonaws.com/downloads/pl/demo-annotated.pdf',
        },
        viewer.current,
      ).then(() => {
        // const { documentViewer } = instance.Core;
        // you can now call WebViewer APIs here...
      });
    }
  }, []);

  return (
    <div className="MyComponent">
      <div className="header">React sample</div>
      <div className="webviewer" ref={viewer} style={{ width: '800px', height: '100vh' }}></div>
    </div>
  );
}

export default App
