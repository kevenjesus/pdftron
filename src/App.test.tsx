import { render } from '@testing-library/react';
import App from "./App"
import { expect, test, vi } from 'vitest';

vi.mock('@pdftron/webviewer', () => ({
    default: vi.fn(() => Promise.resolve({})), // Mocka uma promise vazia que nunca dÃ¡ erro
}));
  

test('O componente App carrega sem erros', () => {
    render(<App />);
    expect(true).toBe(true); // Teste fake que sempre passa
});