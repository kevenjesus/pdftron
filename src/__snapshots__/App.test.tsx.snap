// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Appt esxt > should render 1`] = `
<div>
  <div
    class="MyComponent"
  >
    <div
      class="header"
    >
      React sample
    </div>
    <div
      class="webviewer"
      style="width: 800px; height: 100vh;"
    >
      <apryse-webviewer
        autocreate="true"
        autofocusnoteonannotationselection="true"
        enableannotations="true"
        enableoptimizedworkers="true"
        id="wc-"
        initialdoc="https://pdftron.s3.amazonaws.com/downloads/pl/demo-annotated.pdf"
        path="/public/webviewer/lib"
        preloadpdfworker="true"
        usedownloader="true"
        usesharedworker="false"
        webviewerjsversion="11.1.1"
      />
    </div>
  </div>
</div>
`;
